llvm/lib/Target/DirectX/DXIL.td

```diff
@@ -34,12 +35,42 @@ def BinaryUintCategory : DXILOpCategory<"Binary uint">;
 def UnaryFloatCategory : DXILOpCategory<"Unary float">;
 def ComputeIDCategory : DXILOpCategory<"Compute/Mesh/Amplification shader">;
 
+// ValueTypes specific to DXIL
+// Define Overload value type as an entity with no size and an arbitrary value
+// of 1024 - assuming that all currently defined values are less than 1024
```

Comment:
What happens if LLVM changes to have a value >= 1024? Is it going to break in a easily recognizable way that is obvious to fix? How close are we to that limit?

Reply:
> What happens if LLVM changes to have a value >= 1024? Is it going to break in a easily recognizable way that is obvious to fix? How close are we to that limit?

I have not investigated this. I am looking at a different way to define overload types without the reliance above as I share the same concerns. The current changes do not require these definitions. So, I have pushed a change that deletes these additions from this PR. Thanks!